### DISCUSSION TITLE
Custom content in button toggle

### DISCUSSION THREAD

**Discussion by artreven** [zauberzeug/nicegui#1159] [Q&A] [answered]
### Question

Hi. Thanks a lot for this project, really big advancement compared to streamlit in my opinion.

I was looking for a nice toggle and found this example at Quasar: https://quasar.dev/vue-components/button-toggle#custom-content . I was wondering if it would be possible to replicate the example in NiceGUI using the `.add_slot()` method or any other functionality?

**Comment by falkoschindler** [comment, marked as answer]
Hi @artreven,
This one is tricky! The current implementation of `ui.toggle` doesn't allow setting slot attributes per option. Maybe we can change that in a future release. Until then it's probably easiest to define a custom toggle element like this:

```py
toggle = ui.element('q-btn-toggle')
toggle._props['options'] = [
    {'label': 'One', 'value': 'one', 'slot': 'one'},
    {'label': 'Two', 'value': 'two', 'slot': 'two'},
    {'label': 'Three', 'value': 'three', 'slot': 'three'},
]
with toggle.add_slot('one'):
    ui.tooltip('One!')
with toggle.add_slot('two'):
    ui.tooltip('Two!')
with toggle.add_slot('three'):
    ui.tooltip('Three!')
toggle.on('update:model-value', lambda e: (
    toggle.props(f'model-value={e.args[0]}'),
    ui.notify(f'You selected {e.args[0]}'),
))
```

Note that we need to handle the "update:model-value" event and set `toggle.props(f'model-value={e.args[0]}')` to react on user input. The notification, on the other hand, can be replaced with an arbitrary event handler.
